modules/apps/29-fee/keeper/escrow_test.go

```diff
@@ -55,8 +55,47 @@ func (suite *KeeperTestSuite) TestDistributeFee() {
 				balance = suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), forward, sdk.DefaultBondDenom)
 				suite.Require().Equal(expectedForwardAccBal, balance)
 
-				// check if the refund acc has been refunded the timeoutFee
-				expectedRefundAccBal := refundAccBal.Add(defaultTimeoutFee[0].Add(defaultTimeoutFee[0]))
+				// check if the refund amount is zero
+				expectedRefundAccBal := refundAccBal
+				balance = suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), refundAcc, sdk.DefaultBondDenom)
+				suite.Require().Equal(expectedRefundAccBal, balance)
+
+				// check the module acc wallet is now empty
+				balance = suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), suite.chainA.GetSimApp().IBCFeeKeeper.GetFeeModuleAddress(), sdk.DefaultBondDenom)
+				suite.Require().Equal(sdk.NewCoin(sdk.DefaultBondDenom, sdkmath.NewInt(0)), balance)
+			},
+		},
+		{
+			"success: some refund amount",
+			func() {
+				// set the timeout fee to be greater than recv + ack fee so that the refund amount is non-zero
+				fee.TimeoutFee = sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdkmath.NewInt(400)))
+
+				packetFee = types.NewPacketFee(fee, refundAcc.String(), []string{})
+				packetFees = []types.PacketFee{packetFee, packetFee}
+			},
+			func() {
+				// check if fees has been deleted
+				packetID := channeltypes.NewPacketID(suite.path.EndpointA.ChannelConfig.PortID, suite.path.EndpointA.ChannelID, 1)
+				suite.Require().False(suite.chainA.GetSimApp().IBCFeeKeeper.HasFeesInEscrow(suite.chainA.GetContext(), packetID))
+
+				// check if the reverse relayer is paid
+				expectedReverseAccBal := reverseRelayerBal.Add(defaultAckFee[0]).Add(defaultAckFee[0])
+				balance := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), reverseRelayer, sdk.DefaultBondDenom)
+				suite.Require().Equal(expectedReverseAccBal, balance)
+
+				// check if the forward relayer is paid
+				forward, err := sdk.AccAddressFromBech32(forwardRelayer)
+				suite.Require().NoError(err)
+
+				expectedForwardAccBal := forwardRelayerBal.Add(defaultRecvFee[0]).Add(defaultRecvFee[0])
+				balance = suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), forward, sdk.DefaultBondDenom)
+				suite.Require().Equal(expectedForwardAccBal, balance)
+
+				// check if the refund amount is correct
+				refundCoins := fee.Total().Sub(defaultRecvFee[0]).Sub(defaultAckFee[0])
+				refundCoins = refundCoins.Add(refundCoins...)
```

Comment:
Is there a way we could just use * 2? Like:
```go
refundCoins := fee.Total().Sub(defaultRecvFee[0].Mul(math.NewInt(2)).Sub(defaultAckFee[0].Mul(math.NewInt(2))
```

Could also have:
```go
totalRecvFees := defaultRecvFee[0].Mul(math.NewInt(2))
totalAckFees := defaultAckFee[0].Mul(math.NewInt(2))

...

refundCoins := fee.Total().Sub(totalRecvFees).Sub(totalAckFees)
```

Reply:
I'm actually a bit confused by this test. Why are we doubling the refund coins after subtracting half of the amount spent?

I'm reading:
```go
refund := total() - (ack_spent() / 2) - (recv_spent() / 2)
refund := refund + refund
```
?

Reply:
I updated these to use `mul` like you suggested. And for your question,

```go
refundCoins := fee.Total().Sub(defaultRecvFee[0]).Sub(defaultAckFee[0])
```
refers to the expected amount that should be refunded per packet fee. And then we double it since we send 2 identical packet fee.